libs/api/src/lib/context.ts

```diff
@@ -1014,6 +1018,26 @@ export async function contextFromRequest(
         throw new Error('paymentProvider not found')
       }
 
+      /**
+       * Gradually migrate subscription's payment method.
+       * Mainly used in mutation.public.ts
+       * Requirements written down here https://wepublish.atlassian.net/browse/TSRI-98
+       */
+      if (migrateToTargetPMid) {
+        const updatedSubscription = await prisma.subscription.update({
+          data: {
+            paymentMethodID: migrateToTargetPMid
+          },
+          where: {
+            id: invoice.subscriptionID
```

Comment:
If you get the subscription passed you can use here subscription.id

Reply:
idem